### PR TITLE
Unit test for is_causal Better Transformers (#91900)

### PR DESCRIPTION
**Pull request by jcrousse**
Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/91900

Test Plan:
buck test  :test_transformers -- -r test_train_with_is_causal
buck test mode/opt :test_transformers -- -r test_is_causal_gpu
flake8 test_transformers.py

Differential Revision: D42453642

### CODE REVIEW

File: test/test_transformers.py [hunk 1]
```diff
@@ -1381,16 +1383,76 @@ def test_train_with_is_causal(self, device):
...
+                    # fails if need_weights=False
+                    with self.assertRaisesRegex(RuntimeError, "No available kernel"):
+                        _ = mha(qkv, qkv, qkv, is_causal=True)
+                    # fails with embedding size not multiple of 4
+                    with self.assertRaisesRegex(RuntimeError, "No available kernel"):
+                        qkv_f, mha_f = ones_tensor(S, L, 2), nn.MultiheadAttention(2, H).to(device)
+                        _ = mha_f(qkv_f, qkv_f, qkv_f, need_weights=False, is_causal=True)
+                        torch.cuda.synchronize()
+
+    @unittest.skipIf(not TEST_CUDA or not SM80OrLater or TEST_WITH_ROCM, "CUDA unavailable")
```
Comment (drisspg) [inline test/test_transformers.py:1452]: could you update to use the new https://github.com/pytorch/pytorch/blob/b7cad020b59ec2c9b9d62a1e6ca78c5b53f186a1/test/test_transformers.py#L36
if you are rebased recently enough.

**Review by drisspg — APPROVED** [review]
Looks good besides the one comment

File: test/test_transformers.py (hunk 1 above)
Comment (jcrousse) [reply test/test_transformers.py]: Done!

**Pull request closed without merging**